Utils/src/PmssmProducer.cc

```diff
@@ -15,80 +15,109 @@
-	public:
-		explicit PmssmProducer(const edm::ParameterSet&);
-		~PmssmProducer();
-		static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
-
-	private:
-		virtual void beginJob();
-		virtual void produce(edm::Event&, const edm::EventSetup&);
-		virtual void endJob();
+public:
+  explicit PmssmProducer(const edm::ParameterSet&);
+  ~PmssmProducer();
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  virtual void beginJob();
+  virtual void produce(edm::Event&, const edm::EventSetup&);
+  virtual void endJob();
 		
-		virtual void beginRun(edm::Run&, edm::EventSetup const&);
-		virtual void endRun(edm::Run&, edm::EventSetup const&);
-		virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
-		virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  virtual void endRun(edm::Run&, edm::EventSetup const&);
+  virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
 		
-		void process(std::string line, char delim, std::vector<std::string>& fields);
+  void process(std::string line, char delim, std::vector<std::string>& fields);
+  double getXsec(double pmssmid);
 		
-		// ----------member data ---------------------------
-		edm::GetterOfProducts<LHEEventProduct> getterOfProducts_;
-		bool shouldScan_, debug_;
+  // ----------member data ---------------------------
+  edm::GetterOfProducts<LHEEventProduct> getterOfProducts_;
+  bool shouldScan_, debug_, do_xsec_; std::string _xsecFilename;
 };
 
 PmssmProducer::PmssmProducer(const edm::ParameterSet& iConfig) : 
-	getterOfProducts_(edm::ProcessMatch("*"), this), shouldScan_(iConfig.getParameter<bool>("shouldScan")), debug_(iConfig.getParameter<bool>("debug"))
+  getterOfProducts_(edm::ProcessMatch("*"), this), 
+  shouldScan_(iConfig.getParameter<bool>("shouldScan")), 
+  debug_(iConfig.getParameter<bool>("debug")),
+  do_xsec_(iConfig.getParameter<bool>("do_xsec")),
+  _xsecFilename(iConfig.getParameter<std::string> ("xsecFilename"))
 {
-	callWhenNewProductsRegistered(getterOfProducts_);
-	produces<std::vector<std::string> >("PmssmId");
-	produces<double>("PmssmXsec");	
+  callWhenNewProductsRegistered(getterOfProducts_);
+  produces<double >("PmssmId");
+  produces<double>("PmssmXsec");	
 }
 
 PmssmProducer::~PmssmProducer()
 {
 	
-	// do anything here that needs to be done at desctruction time
-	// (e.g. close files, deallocate resources etc.)
-	
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+double PmssmProducer::getXsec(double pmssmid)
+{
+  double xsec = -1;
+  std::ifstream fileInput;
+  std::string line;//use a sprintf with pmssmid
+  char search [30];
+  sprintf (search, "%d", int(pmssmid));
```

Comment:
This function is never called in TreeMaker because the config boolean "do_xec" is set to false. This function is not called in TreeMaker because I modified the existing WeightProducer modules to get the cross section to be seamless with the way the other FastSim xsecs are obtained (the map is read out once by weightproducer, etc, by the existing function). The function referred to is only there because it is part of the  official EDProducer being passed around to the SUSY teams doing a pMSSM interpretation.